app/src/main/java/org/dicekeys/app/fragments/dicekey/SoloKeyFragment.kt

```diff
@@ -1,21 +1,230 @@
+import android.content.IntentFilter
+import android.hardware.usb.UsbDevice
+import android.hardware.usb.UsbManager
 import android.os.Bundle
+import android.os.CountDownTimer
 import android.view.View
+import androidx.core.view.isVisible
+import androidx.core.widget.doAfterTextChanged
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.Observer
+import com.dicekeys.fidowriter.ListOfWritableUsbFidoKeys
 import dagger.hilt.android.AndroidEntryPoint
+import org.dicekeys.api.DerivationRecipe
+import org.dicekeys.api.derivationRecipeTemplates
 import org.dicekeys.app.AppFragment
 import org.dicekeys.app.R
 import org.dicekeys.app.databinding.SolokeyFragmentBinding
 import org.dicekeys.app.viewmodels.DiceKeyViewModel
+import org.dicekeys.crypto.seeded.Secret
 
 @AndroidEntryPoint
 class SoloKeyFragment: AppFragment<SolokeyFragmentBinding>(R.layout.solokey_fragment) {
 
+    private lateinit var deviceList: ListOfWritableUsbFidoKeys
+    private lateinit var permissionIntent: android.app.PendingIntent
     private lateinit var viewModel: DiceKeyViewModel
+    private val INTENT_ACTION_USB_PERMISSION_EVENT = "org.dicekeys.intents.USB_PERMISSION_EVENT"
+    private var isWriteUnderway: Boolean = false
+    private var derivationRecipe = MutableLiveData<DerivationRecipe>()
+    private var sequenceNumber = MutableLiveData<Int>(1)
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
 
         viewModel = getDiceKeyRootFragment().viewModel
+
+        binding.vm = viewModel
+
+        /*Get selected derivationRecipe template*/
+        derivationRecipe.value = derivationRecipeTemplates.get(1)
+
+        binding.tvRecipeJson.text = derivationRecipe.value!!.derivationOptionsJson
+        binding.btnDown.setOnClickListener { sequencUpDown(false) }
+        binding.btnUp.setOnClickListener { sequencUpDown(true) }
+        permissionIntent = android.app.PendingIntent.getBroadcast(context, 0, Intent(INTENT_ACTION_USB_PERMISSION_EVENT), 0)
+        deviceList = ListOfWritableUsbFidoKeys(
+                context?.getSystemService(android.content.Context.USB_SERVICE) as UsbManager,
+                permissionIntent
+        )
+
+        /*Observer sequence number changes & update derivationRecipe object*/
+        sequenceNumber.observe(viewLifecycleOwner, Observer { intValue ->
+            derivationRecipe.value = DerivationRecipe(derivationRecipeTemplates.get(1), intValue)
+        })
+
+        /*Observer derivation recipe change and update password, derivationOptionsJson */
+        derivationRecipe.observe(viewLifecycleOwner, Observer { value ->
+            val seed = Secret.deriveFromSeed(viewModel.diceKey.seed, value.derivationOptionsJson).secretBytes
+            binding.tvSeedHexString.text = (seed.joinToString(separator = "") { String.format("%02x", (it.toInt() and 0xFF)) })
+            binding.tvRecipeJson.text = value.derivationOptionsJson
```

Comment:
Here you have to set the text to tvRecipeJson, use a viewmodel and have the text like this:
```
...
android:text="@{vmSolo. derivationRecipe.derivationOptionsJson}"
...
```